packages/@lwc/synthetic-shadow/src/faux-shadow/portal.ts

```diff
@@ -5,13 +5,17 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
  */
 import { isUndefined, forEach, defineProperty, isTrue } from '@lwc/shared';
-import { getInternalChildNodes } from './node';
-import { compareDocumentPosition } from '../env/node';
-import { setShadowRootResolver, ShadowRootResolver, getShadowRootResolver } from './shadow-root';
+import { childNodesGetter, compareDocumentPosition } from '../env/node';
+import { MutationObserver, MutationObserverObserve } from '../env/mutation-observer';
+import {
+    setShadowRootResolver,
+    ShadowRootResolver,
+    getShadowRootResolver,
+    isHostElement,
+} from './shadow-root';
 import { setShadowToken, getShadowToken } from './shadow-token';
 
 const MO = MutationObserver;
```

Comment:
Avoid caching the MutationObserver in a local const. I had issues where `const MO = MutationObserver` was resolving the value from the window instead of the imported value.

Reply:
I don't understand this.

Reply:
I had an issue where the following code would resolve MutationObserver on window in the const assignment. So it would get the polyfilled MO.
```
import { MutationObserver, MutationObserverObserve } from '../env/mutation-observer';
const MO = MutationObserver;
```

Reply:
ok, yes... the order might be wrong... better to import it from the cached env file.